### DISCUSSION TITLE
Naming of backup files created by console "./craft db/backup" by DB status not Craft's version

### DISCUSSION THREAD

**Discussion by hiasl** [craftcms/cms#9733] [Ideas] [answered]
When using Craft's console command "./craft db/backup" to dump the DB, filenames are created by concatting
`<project-name>-<timestamp>-<version>.<extension>`
e.g.
`myproject--2021-08-17-132930--v3.7.8.sql.zip`

The version number is not taken from the state of the DB but from the state of Craft's installed version.

In other words:
- You have Craft running with version X
- You do a `composer install` or `composer update` to version Y, but do not complete the migrations yet, so DB is still in state X
- You call `./craft db/backup`
- The filename created will reflect version Y not X, although the backup can not be used with version Y. 

IMHO this is not good. You should be able to choose the correct dump for a certain Craft version by reading its filename.
Therefore I think Craft should use the version from the "info" table for naming backups, if available.

And if you wonder, why I backup AFTER `composer install` or `composer update`. Because in some cases Craft's sources are not yet available and I need to backup after installing the current version, which might be newer than the DB itself.

**Comment by brandonkelly** [comment, marked as answer]
Agree that this is a bit awkward. I’ll consider changing it for Craft 4.

**Reply by brandonkelly** [reply]
Done for Craft 4 (#10102).
